src/Promitor.Agents.ResourceDiscovery/Program.cs

```diff
@@ -57,24 +67,43 @@ private static ServerConfiguration GetServerConfiguration(IConfiguration configu
             return serverConfiguration;
         }
 
-        private static IConfiguration BuildConfiguration(string[] args)
+        private static IConfiguration BuildConfiguration(string[] args, string configurationFolder)
         {
-            var configurationFolder = Environment.GetEnvironmentVariable(EnvironmentVariables.Configuration.Folder);
-            if (string.IsNullOrWhiteSpace(configurationFolder))
-            {
-                throw new Exception("Unable to determine the configuration folder");
-            }
-
             IConfigurationRoot configuration =
                 new ConfigurationBuilder()
                     .SetBasePath(Directory.GetCurrentDirectory())
                     .AddCommandLine(args)
-                    .AddYamlFile($"{configurationFolder}/runtime.yaml", optional: false, reloadOnChange: true)
-                    .AddYamlFile($"{configurationFolder}/resource-discovery-declaration.yaml", optional: false, reloadOnChange: true)
+                    .AddYamlFile($"{configurationFolder}/{RuntimeConfigFilename}", optional: false, reloadOnChange: true)
+                    .AddYamlFile($"{configurationFolder}/{ResourceDiscoveryFilename}", optional: false, reloadOnChange: true)
                     .AddEnvironmentVariables("PROMITOR_")
                     .Build();
 
             return configuration;
         }
+
+        private static ExitStatus? ValidateConfigurationExists(string configurationFolder)
+        {
+            if (string.IsNullOrWhiteSpace(configurationFolder))
+            {
+                Log.Logger.Fatal($"Unable to determine the configuration folder. Please ensure that the '{EnvironmentVariables.Configuration.Folder}' environment variable is set");
+                return ExitStatus.ConfigurationFolderNotSpecified;
+            }
+
+            var runtimeConfigPath = Path.Combine(configurationFolder, RuntimeConfigFilename);
```

Comment:
This can potentially get out of sync with `AddYamlFile` steps, should we move them to dedicated method or so? Might be overkill though.

Thoughts?

Reply:
Yeah - I think that makes sense. I'll try to rework it, and maybe share the code between the two agents.

Reply:
@tomkerkhove I've rejigged this by adding a new extension method, `AddRequiredYamlFile()` that just wraps the built in one. This means we can check if the file exists and throw a custom exception that can be caught so we can write out the right error message and set the correct exit code. The reason I did this is that I think `AddYamlFile()` just throws a `FileNotFoundException`, which we could catch, but it could be thrown for a whole load of other reasons.

Anyhow - let me know what you think.